modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala

```diff
@@ -304,10 +297,10 @@ class SeqexecEngine(httpClient: Client[IO], settings: SeqexecEngine.Settings, sm
 
   def startQueue(q: EventQueue, qid: QueueId, clientId: ClientID): IO[Either[SeqexecFailure, Unit]] = q.enqueue1(
     Event.modifyState[executeEngine.ConcreteTypes](executeEngine.get.flatMap{ st => {
-      (EngineState.queues ^|-? index(qid)).getOption(st).map {
-        _.status(st) match {
-          case BatchExecState.Idle     => ((EngineState.queues ^|-? index(qid) ^|-> ExecutionQueue.cmdState).set(BatchCommandState.Run(clientId)) >>> {(_, ())}).toHandle *> runQueue(qid, clientId)
-          case BatchExecState.Stopping => ((EngineState.queues ^|-? index(qid) ^|-> ExecutionQueue.cmdState).set(BatchCommandState.Run(clientId)) >>> {(_, ())}).toHandle
+      queueO(qid).getOption(st).map { q =>
+        q.status(st) match {
+          case BatchExecState.Idle     => (cmdStateO(qid).set(BatchCommandState.Run(clientId)) >>> execStateO(qid).set(q.status(st)) >>> {(_, ())}).toHandle *> runQueue(qid, clientId)
```

Comment:
I'm setting the state here in `startQueue` and on `stopQueue` but I'm not getting these updates on the client. I must be missing something

Reply:
It is because the sequence states are updated inside `runQueque`. Your call to `q.status` should be after the call to `runQueue` to see the changes.